lightning/src/ln/channel.rs

```diff
@@ -1895,6 +1895,16 @@ impl<Signer: Sign> Channel<Signer> {
 			return Err(ChannelError::Close("Minimum confirmation depth must be at least 1".to_owned()));
 		}
 
+		if let Some(ty) = &msg.channel_type {
+			if *ty != self.channel_type {
+				return Err(ChannelError::Close("Channel Type in accept_channel didn't match the one sent in open_channel.".to_owned()));
+			}
+		} else if their_features.supports_channel_type() {
```

Comment:
IIUC we might have a peer supporting channel type (feature bits 44/45) but not announcing any `channel_type` because they don't support any channel feature. What's the state of mainstream implementations, they all support static_key at least ? 

Maybe we can implement the spec suggestion : "The receiving node MAY fail the channel if `option_channel_type` was negotiated but the message doesn't include a `channel_type`". That would be a first step deprecating channel type detection from Init message (and I think we can deprecate such logic once `channel_type` is well-deployed, as it's doesn't constrain to force-close alive channel such as when deprecating commitment format). Maybe too early.

Reply:
> IIUC we might have a peer supporting channel type (feature bits 44/45) but not announcing any channel_type because they don't support any channel feature. What's the state of mainstream implementations, they all support static_key at least ?

No, the point of the feature bit is they *have* to announce a channel type field/TLV, though it can be empty.

> The receiving node MAY fail the channel if option_channel_type was negotiated but the message doesn't include a channel_type". 

Two things: a) this is about `open_channel` not `accept_channel`, and (b) that's how I had this PR initially, but that means new-LDK won't be able to accept a channel from current-LDK, which is not ideal.